plugins/modules/sonic_qos_pfc.py

```diff
@@ -0,0 +1,237 @@
+author: "Shade Talabi (@stalabi1)"
+options:
+  config:
+    description:
+      - QoS PFC configuration
+    type: dict
+    suboptions:
+      counter_poll:
+        description:
+          - Enable or disable use of flex-counters for PFC watchdog
+        type: bool
+        default: True
+      poll_interval:
+        description:
+          - Polling interval for PFC watchdog
+          - Range 100-3000
+        type: int
+  state:
+    description:
+      - The state of the configuration after module completion.
+    type: str
+    choices:
+    - merged
+    - deleted
+    - overridden
+    - replaced
+    default: merged
+"""
+EXAMPLES = """
+# Using Merged
+#
+# Before state:
+# -------------
+#
+# sonic# show priority-flow-control watchdog
+#
+# Watchdog Summary
+# ----------------
+# Polling Interval:   : Not Available
+# Flex Counters:      : enabled
+
+- name: Merge QoS PFC configurations
+  dellemc.enterprise_sonic.sonic_qos_pfc:
+    config:
+      counter_poll: True
+      poll_interval: 150
+    state: merged
+
+# After state:
+# ------------
+#
+# sonic# show priority-flow-control watchdog
+#
+# Watchdog Summary
+# ----------------
+# Polling Interval:   : 150
+# Flex Counters:      : enabled
+#
+#
+# Using Replaced
+#
+# Before state:
+# -------------
+#
+# sonic# show priority-flow-control watchdog
+#
+# Watchdog Summary
+# ----------------
+# Polling Interval:   : 150
+# Flex Counters:      : enabled
+
+- name: Replace QoS PFC configurations
+  dellemc.enterprise_sonic.sonic_qos_pfc:
+    config:
+      poll_interval: 365
+    state: replaced
+
+# After state:
+# ------------
+#
+# sonic# show priority-flow-control watchdog
+#
+# Watchdog Summary
+# ----------------
+# Polling Interval:   : 365
+# Flex Counters:      : enabled
+#
+#
+# Using Overridden
+# Before state:
+# -------------
+#
+# sonic# show priority-flow-control watchdog
+#
+# Watchdog Summary
+# ----------------
+# Polling Interval:   : 365
+# Flex Counters:      : enabled
+
+- name: Override QoS PFC configurations
+  dellemc.enterprise_sonic.sonic_qos_pfc:
+    config:
+      counter_poll: False
+      poll_interval: 400
+    state: overridden
+
+# After state:
+# ------------
+#
+# sonic# show priority-flow-control watchdog
+#
+# Watchdog Summary
+# ----------------
+# Polling Interval:   : 400
+# Flex Counters:      : disabled
+#
+#
+# Using deleted
+#
+# Before state:
+# -------------
+#
+# sonic# show priority-flow-control watchdog
+#
+# Watchdog Summary
+# ----------------
+# Polling Interval:   : 400
+# Flex Counters:      : disabled
+
+- name: Delete QoS PFC configurations
+  dellemc.enterprise_sonic.sonic_qos_pfc:
+    config:
+      counter_poll: False
+      poll_interval: 400
+    state: deleted
+
+# After state:
+# ------------
+#
+# sonic# show priority-flow-control watchdog
+#
+# Watchdog Summary
+# ----------------
+# Polling Interval:   : Not Available
+# Flex Counters:      : enabled
+"""
+RETURN = """
```

Comment:
In the block below:
- Please change all instances of "model" to "module".
- Please change all instances of "of the parameters above" to "as the parameters above".